MonoGame.Framework/Graphics/Texture2D.DirectX.cs

```diff
@@ -86,76 +88,79 @@ private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat
-            desc.Width = levelWidth;
-            desc.Height = levelHeight;
-            desc.MipLevels = 1;
-            desc.ArraySize = 1;
-            desc.Format = SharpDXHelper.ToFormat(_format);
-            desc.BindFlags = BindFlags.None;
-            desc.CpuAccessFlags = CpuAccessFlags.Read;
-            desc.SampleDescription.Count = 1;
-            desc.SampleDescription.Quality = 0;
-            desc.Usage = ResourceUsage.Staging;
-            desc.OptionFlags = ResourceOptionFlags.None;
-
-            // Save sampling description.
-            _sampleDescription = desc.SampleDescription;
+            if (cachedStagingTexture == null)
+            {
+                var desc = new Texture2DDescription();
+                desc.Width = levelWidth;
+                desc.Height = levelHeight;
+                desc.MipLevels = 1;
+                desc.ArraySize = 1;
+                desc.Format = SharpDXHelper.ToFormat(_format);
+                desc.BindFlags = BindFlags.None;
+                desc.CpuAccessFlags = CpuAccessFlags.Read;
+                desc.SampleDescription.Count = 1;
+                desc.SampleDescription.Quality = 0;
+                desc.Usage = ResourceUsage.Staging;
+                desc.OptionFlags = ResourceOptionFlags.None;
+
+                // Save sampling description.
+                _sampleDescription = desc.SampleDescription;
+
+                cachedStagingTexture = new SharpDX.Direct3D11.Texture2D(GraphicsDevice._d3dDevice, desc);
+            }
 
             var d3dContext = GraphicsDevice._d3dContext;
-            using (var stagingTex = new SharpDX.Direct3D11.Texture2D(GraphicsDevice._d3dDevice, desc))
+
+            lock (d3dContext)
             {
-                lock (d3dContext)
-                {
-                    var subresourceIndex = CalculateSubresourceIndex(arraySlice, level);
+                var subresourceIndex = CalculateSubresourceIndex(arraySlice, level);
 
-                    // Copy the data from the GPU to the staging texture.
-                    var elementsInRow = rect.Width;
-                    var rows = rect.Height;
-                    var region = new ResourceRegion(rect.Left, rect.Top, 0, rect.Right, rect.Bottom, 1);
-                    d3dContext.CopySubresourceRegion(GetTexture(), subresourceIndex, region, stagingTex, 0);
+                // Copy the data from the GPU to the staging texture.
+                var elementsInRow = rect.Width;
+                var rows = rect.Height;
+                var region = new ResourceRegion(rect.Left, rect.Top, 0, rect.Right, rect.Bottom, 1);
+                d3dContext.CopySubresourceRegion(GetTexture(), subresourceIndex, region, cachedStagingTexture, 0);
+
+                // Copy the data to the array.
+                DataStream stream = null;
+                try
+                {
+                    var databox = d3dContext.MapSubresource(cachedStagingTexture, 0, MapMode.Read, MapFlags.None, out stream);
```

Comment:
Yeah, before when we were making a new Tex2D it wouldn't matter if Unmap was never called on it it looks like. But now that we are caching, we have to unmap properly or else shit will get wacky